app/src/main/java/cloudcity/util/CellUtil.java

```diff
@@ -68,27 +71,33 @@ public static CellInformation findRegisteredCell(@NonNull List<CellInformation>
         return measurements;
     }
 
-    public static MeasurementsModel getRegisteredCellInformationUpdatedBySignalStrengthInformation(DataProvider dp) {
-        List<CellInformation> cellsInfo = dp.getCellInformation();
-        List<CellInformation> signalInfo = dp.getSignalStrengthInformation();
-
-        CellInformation currentCell = CellUtil.findRegisteredCell(cellsInfo);
-        CellInformation currentSignal = null;
-        // After testing on a real phone, apparently there's only one SignalStrengthInformation
-        // CellInformation member in the list, so lets just take the first one if it's there
-        if (!signalInfo.isEmpty()) {
-            currentSignal = signalInfo.get(0);
-        }
+    public static MeasurementsModel getRegisteredCellInformationUpdatedBySignalStrengthInformation(@NonNull DataProvider dp) {
```

Comment:
no. the `@NonNull` is to prevent you from passing null in there at compile-time; since the `DataProvider` is out of our control, as it's part of the OMNT codebase, it **does** become randomly null at times.

So the jist of it is - the method doesn't let you pass nulls at compile time, but can't force you to never pass null in there at run-time (since we're not in Kotlin) so we need to handle the case when a `null` DataProvider gets passed in here as well.